feat(unity): add thread-safety note to Hooks (#4944)
getsentry/sentry-docs: src/platforms/common/configuration/options.mdx

These options can be used to hook the SDK in various ways to customize the reporting of events.

<PlatformSection supported={["unity"]}>

The callbacks you specify as hooks will be called on the thread where the event happened. Therefore, you may only use

> thread-safe APIs and only use UI-specific APIs after you've checked that you're on the UI thread.

Reviewer: ```suggestion
thread-safe APIs and only use Unity-specific APIs after you've checked that you're on the UI thread.
```

> The callbacks you specify as hooks will be called on the thread where the event happened. Therefore, you may only use
> thread-safe APIs and only use Unity-specific APIs after you've checked that you're on the UI thread.

Reviewer 2: Iv'e made some changes to simplify the language a bit

```suggestion
The callbacks you set as hooks will be called on the thread where the event happened. So you can only use
thread-safe APIs and only use Unity-specific APIs after you've checked that you're on the UI thread.
```
